servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -0,0 +1,432 @@
+      registerBlockStoreEventListener((BlockStoreEventListener) mEvictor);
+    }
+  }
+
+  @Override
+  public long lockBlock(long userId, long blockId) throws IOException {
+    return mLockManager.lockBlock(userId, blockId, BlockLockType.READ);
+  }
+
+  @Override
+  public void unlockBlock(long lockId) throws IOException {
+    mLockManager.unlockBlock(lockId);
+  }
+
+  @Override
+  public void unlockBlock(long userId, long blockId) throws IOException {
+    mLockManager.unlockBlock(userId, blockId);
+  }
+
+  @Override
+  public BlockWriter getBlockWriter(long userId, long blockId) throws IOException {
+    TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
+    return new LocalFileBlockWriter(tempBlockMeta);
+  }
+
+  @Override
+  public BlockReader getBlockReader(long userId, long blockId, long lockId) throws IOException {
+    mLockManager.validateLockId(userId, blockId, lockId);
+    BlockMeta blockMeta = mMetaManager.getBlockMeta(blockId);
+    return new LocalFileBlockReader(blockMeta);
+  }
+
+  @Override
+  public TempBlockMeta createBlockMeta(long userId, long blockId, BlockStoreLocation location,
+      long initialBlockSize) throws IOException {
+    TempBlockMeta tempBlock;
+    mEvictionLock.readLock().lock();
+    try {
+      tempBlock = createBlockMetaNoLock(userId, blockId, location, initialBlockSize);
+    } finally {
+      mEvictionLock.readLock().unlock();
+    }
+    return tempBlock;
+  }
+
+  @Override
+  public BlockMeta getBlockMeta(long userId, long blockId, long lockId) throws IOException {
+    mLockManager.validateLockId(userId, blockId, lockId);
+    return mMetaManager.getBlockMeta(blockId);
+  }
+
+  @Override
+  public void commitBlock(long userId, long blockId) throws IOException {
+    mEvictionLock.readLock().lock();
+    TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
+    try {
+      commitBlockNoLock(userId, blockId, tempBlockMeta);
+      for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
+        listener.onCommitBlock(userId, blockId, tempBlockMeta.getBlockLocation());
+      }
+    } finally {
+      mEvictionLock.readLock().unlock();
+    }
+  }
+
+  @Override
+  public void abortBlock(long userId, long blockId) throws IOException {
+    mEvictionLock.readLock().lock();
+    try {
+      abortBlockNoLock(userId, blockId);
+      for (BlockStoreEventListener listener : mBlockStoreEventListeners) {
+        listener.onAbortBlock(userId, blockId);
+      }
+    } finally {
+      mEvictionLock.readLock().unlock();
+    }
+  }
+
+  @Override
+  public void requestSpace(long userId, long blockId, long moreBytes) throws IOException {
+    // TODO: Change the lock to read lock and only upgrade to write lock if necessary
+    mEvictionLock.writeLock().lock();
+    try {
+      TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
+      BlockStoreLocation location = tempBlockMeta.getBlockLocation();
```

Comment:
Minor style comment, this variable only gets used once and the call is pretty clear what it is getting.